rmgpy/molecule/resonance.py

```diff
@@ -341,17 +389,187 @@ def generate_isomorphic_isomers(mol):
 
     return isomorphic_isomers
 
-def populate_resonance_generation_algorithm():
+def generateClarStructures(mol):
     """
-    A list with the current set of resonance generation algorithms.
+    Generate Clar structures for a given molecule.
+
+    Returns all Clar structures as a list.
     """
-    algorithms = (
-        generateAdjacentResonanceIsomers,
-        generateLonePairRadicalResonanceIsomers,
-        generateN5dd_N5tsResonanceIsomers,
-        generateKekulizedResonanceIsomers,
-        generateAromaticResonanceIsomers,
-    )
+    cython.declare(output=list, molList=list, newmol=Molecule, SSSR=list, bonds=list, solution=list,
+                   y=list, x=list, index=cython.int, bond=Bond, ring=list)
+
+    if not mol.isCyclic():
+        return []
+
+    output = clarOptimization(mol)
+
+    molList = []
+
+    for newmol, SSSR, bonds, solution in output:
```

Comment:
Ok, good to know. I used caps because most usages of getSmallestSetOfSmallestRings() assigns the result to SSSR. I'm fine with changing to lower case sssr though.